src/Telegram.Bot/Types/User.cs

```diff
@@ -44,5 +45,60 @@ public class User
         /// </summary>
         [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
         public string LanguageCode { get; set; }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj) => Equals(obj as User);
+
+        /// <inheritdoc />
+        public bool Equals(User other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+
+            return Id == other.Id &&
```

Comment:
Specifically only used Id to compare, so that a user still compares the same, if they change their name at any time during conversation with the bot